alsi/mappings.py

```diff
@@ -0,0 +1,102 @@
+from enum import Enum
+from typing import Union
+from .exceptions import InvalidCountryException
+
+
+def retrieve_country(input: Union["Area", str]) -> "Area":
+
+    if isinstance(input, Area):
+        return input
+    if isinstance(input, str):
+        if Area.has_member(input.upper()):
+            return Area[input.upper()]
+
+        check_country = [
+            area
+            for area in Area
+            if area._country_name.lower() == input.lower()
+        ]
+
+        if check_country:
+            return check_country[0]
+        else:
+            raise InvalidCountryException("Invalid Country")
+
+
+class Area(Enum):
+    def __new__(cls, *args, **kwargs):
+        obj = object.__new__(cls)
+        obj._value_ = args[0]
+        return obj
+
+    def __init__(self, _: str, country_name: str):
+
+        self._country_name = country_name
+
+    @property
+    def name(self) -> str:
+        return self._country_name
+
+    @property
+    def code(self) -> str:
+        return self.value
+
+    @classmethod
+    def has_member(cls, member) -> bool:
+        return member in cls.__members__
+
+    def __str__(self) -> str:
+        return self.value
+
+    BE = (
```

Comment:
Why split it?

```python
BE = ("BE", "Belgium")
```

This is fine since it is belowe 80 characters length. Does `black` format it like this?

Reply:
Yes. Black actually split it.

Reply:
Ha, interesting. OK!